app/scripts/modules/core/src/pipeline/config/PipelineRegistry.spec.ts

```diff
@@ -182,6 +183,51 @@ describe('PipelineRegistry: API', function() {
     );
   });
 
+  describe('preconfigured stage', function() {
+    beforeEach(mock.inject());
+
+    const makeStage = () => ({ key: 'job', alias: 'preconfiguredJob' } as IStageTypeConfig);
```

Comment:
am I reading correctly that folks writing stage plugins will still have to add `alias: 'preconfiguredJob'` to their plugin stage config? if so, any idea how ok/terrible it would be to handle that for them in `registerPreconfiguredJobStage()`?

Reply:
There's a helper `buildPreconfiguredJobStage` that builds a preconfigured job stage from the PreconfigureJob metadata (the shape that Gate returns).  So in actual code (not tests) it would look more like this:

```
export const treasureDeployStage: IStageTypeConfig = {
  ...buildPreconfiguredJobStage({
    uiType: 'CUSTOM',
    type: 'treasureDeploy',
    label: `Treasure Deploy with UI (EXPERIMENTAL)`,
    noUserConfigurableFields: false,
    description: 'Publishes static assets to treasure,
    producesArtifacts: false,
  }),
  component: TreasureStageConfig, // stage config
  // executionDetailsSections: [PreconfiguredJobExecutionDetails, ExecutionDetailsTasks],
  executionDetailsSections: [ExecutionDetailsTasks],
  validateFn: validate,
};
```
As I'm looking at this code now, I see that a lot of it is duplicated from the backend orca registration.  I think I'll refactor this a bit more in hopes of simplifying it further for devs.